pkg/tsdb/azuremonitor/azure-resource-graph-datasource.go

```diff
@@ -0,0 +1,257 @@
+		}
+
+		queryJSONModel := argJSONQuery{}
+		err = json.Unmarshal(queryBytes, &queryJSONModel)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode the Azure Resource Graph query object from JSON: %w", err)
+		}
+
+		azureResourceGraphTarget := queryJSONModel.AzureResourceGraph
+		azlog.Debug("AzureResourceGraph", "target", azureResourceGraphTarget)
+
+		resultFormat := azureResourceGraphTarget.ResultFormat
+		if resultFormat == "" {
+			resultFormat = "table"
+		}
+
+		params := url.Values{}
+		rawQuery, err := KqlInterpolate(query, timeRange, azureResourceGraphTarget.Query)
+
+		if err != nil {
+			return nil, err
+		}
+		params.Add("query", rawQuery)
+
+		azureResourceGraphQueries = append(azureResourceGraphQueries, &AzureResourceGraphQuery{
+			RefID:        query.RefID,
+			ResultFormat: resultFormat,
+			Model:        query.Model,
+			Params:       params,
+			Target:       params.Encode(),
+		})
+	}
+
+	return azureResourceGraphQueries, nil
+}
+
+func (e *AzureResourceGraphDatasource) executeQuery(ctx context.Context, query *AzureResourceGraphQuery,
+	timeRange plugins.DataTimeRange) backend.DataResponse {
+	queryResult := backend.DataResponse{}
+
+	query.Params.Add("api-version", argAPIVersion)
+
+	queryResultErrorWithExecuted := func(err error) backend.DataResponse {
+		queryResult = backend.DataResponse{Error: err}
+		frames := data.Frames{
+			&data.Frame{
+				RefID: query.RefID,
+				Meta: &data.FrameMeta{
+					ExecutedQueryString: query.Params.Get("query"),
+				},
+			},
+		}
+		queryResult.Frames = frames
+		return queryResult
+	}
+
+	reqBody := fmt.Sprintf("{'subscriptions':['%s'], 'query':'%s'}",
+		strings.Join(query.Model.Get("subscriptions").MustStringArray(), "','"),
+		query.Model.Get("azureResourceGraph").Get("query").MustString())
+	req, err := e.createRequest(ctx, e.dsInfo, reqBody)
+
+	if err != nil {
+		queryResult.Error = err
+		return queryResult
+	}
+
+	req.URL.Path = path.Join(req.URL.Path, argQueryProviderName)
+	req.URL.RawQuery = query.Params.Encode()
+
+	span, ctx := opentracing.StartSpanFromContext(ctx, "azure resource graph query")
+	span.SetTag("target", query.Target)
+	span.SetTag("from", timeRange.From)
+	span.SetTag("until", timeRange.To)
+	span.SetTag("datasource_id", e.dsInfo.Id)
+	span.SetTag("org_id", e.dsInfo.OrgId)
+
+	defer span.Finish()
+
+	if err := opentracing.GlobalTracer().Inject(
+		span.Context(),
+		opentracing.HTTPHeaders,
+		opentracing.HTTPHeadersCarrier(req.Header)); err != nil {
+		return queryResultErrorWithExecuted(err)
+	}
+
+	azlog.Debug("AzureResourceGraph", "Request ApiURL", req.URL.String())
+	res, err := ctxhttp.Do(ctx, e.httpClient, req)
+	if err != nil {
+		return queryResultErrorWithExecuted(err)
+	}
+
+	argResponse, err := e.unmarshalResponse(res)
+	if err != nil {
+		return queryResultErrorWithExecuted(err)
+	}
+
+	frame, err := LogTableToFrame(&argResponse.Data)
+	if err != nil {
+		return queryResultErrorWithExecuted(err)
+	}
+
```

Comment:
Could you also add this? That way the raw query will be displayed in the Query inspector in the Query editor. It's very helpful when debugging a query.
```go
	if (frame.Meta == nil) {
		frame.Meta = &data.FrameMeta{}
	}
	frame.Meta.ExecutedQueryString = req.URL.RawQuery
```